### ISSUE TITLE
An option to disable animations

### ISSUE THREAD

**Issue by gf-db** [sqlitebrowser/sqlitebrowser#2780]
#### What did you do?
Expand or fold any subtree in the Database Structure tab. Needs an open database with at least one non-empty table/index/view/trigger.

#### What did you expect to see?
A view that is immediately changed to show the desired result (expanded/folded subtree).

#### What did you see instead?
Forced slow sliding animation.

Useful extra information
-------------------------
A simple fix would involve adding a "Disable animations" switch in the DB Browser settings.
A more elaborate option would be to read some OS-specific settings, to figure out if the user wants animations or not.
Personally, I think the elaborate approach is prone to bugs and unintuitive behaviour, and a simple fix would suffice.

OS: Windows 7 x64 SP1 (shouldn't matter)
DB4S: 3.12.2

**Comment by chrisjlocke** [comment]
I've used DB4S for many years, and can't say I've ever noticed this before.  On a large database (30+ tables) collapsing and expanding the table leg doesn't take an inordinate amount of time.  For the avoidance of doubt, this is just a quick demo of 22 tables.

https://user-images.githubusercontent.com/7872218/126696569-e010c3a1-d0a0-44f7-8aeb-ce47153008ec.mp4

Is this the sort of delay you're experiencing, or is your delay far greater than this?

**Comment by mgrojo** [comment]
Maybe current Qt versions are not very well tested/working under Windows 7.

**Comment by gf-db** [comment]
Yes, that video is representative of what I'm seeing.

**Comment by justinclift** [comment]
Um... this might be a silly question on my part, but I'm not personally understanding the problem with the animation.  It doesn't _seem_ to be doing the wrong thing, or taking a huge amount of time.

Am I misunderstanding? :smile:

**Comment by gf-db** [comment]
My reaction is certainly not universal, but I don't think it's limited personally to me, either. I'll try to explain.
The animation feels wrong and annoying because it introduces an artificial and unnecessary delay in your workflow. Whenever I click a plus/minus icon in the UI I'm doing it to arrive at a certain UI state: an expanded/folded list. Today's PCs are pretty capable of delivering appropriate result on the next screen refresh frame, or maybe if you're unlucky, the frame after that. Instead of that the software forces me to sit through the animation that serves no useful purpose. It might be useful as an illustration for absolute novices who don't have a concept of a collapsed list for the first three times they click it, but anyone using a db viewer interface is well past that stage.
If a database takes 300 ms to return a result, then that's fine: it is doing exactly what you asked it when you asked it to the best of its ability. But when UI takes 300 ms to do what should have taken 300 us, for me, honestly, that's infuriating. Such a delay might seem insignificant, but a hundred of clicks and you've lost half a minute of your life for nothing. Then multiply that by the number of users.

**Comment by justinclift** [comment]
No worries, understood.  It's a pretty edge case thing, so it's pretty _unlikely_ we'd spend a lot of time figuring out how to make this happen ourselves.

That being said, our application is built on a GUI toolkit called Qt (https://www.qt.io).  Qt has a bunch of variables that you can set in your environment to enable or disable certain behaviour for applications.  So, my best guess is to do some searching on the Qt website (and probably general search engine searching too), to see if there's something for animations for Qt.

I'd say it's probably a 50/50 chance, but it doesn't hurt to spend some time looking. :smile:

**Comment by gf-db** [comment]
Nope, there is no env var for this. It is controlled by the "animated" property of the "dbTreeWidget" QTreeView in src/MainWindow.ui, but I'd guess if you simply tear it off then those who liked the animations will come to complain. Will you accept a patch with a setting for this in Preferences -> General?

**Comment by mgrojo** [comment]
Can you check whether that effect can be disabled globally under Windows, before making that patch?
https://lifehacker.com/how-to-disable-animations-in-windows-and-make-it-feel-1728734414

**Comment by gf-db** [comment]
I had both "reduce animation" turned on somehow and all but two ("Show contents while dragging" and "Smooth edges of screen fonts") visual effects unchecked. No effect on Qt.
